wgpu-core/src/hub.rs

```diff
@@ -149,46 +149,59 @@ impl<T, I: id::TypedId> ops::IndexMut<id::Valid<I>> for Storage<T, I> {
 impl<T, I: id::TypedId> Storage<T, I> {
     pub(crate) fn contains(&self, id: I) -> bool {
         let (index, epoch, _) = id.unzip();
-        match self.map[index as usize] {
-            Element::Vacant => false,
-            Element::Occupied(_, storage_epoch) | Element::Error(storage_epoch, ..) => {
-                epoch == storage_epoch
-            }
+        match self.map.get(index as usize) {
+            Some(element) => match *element {
+                Element::Vacant => false,
+                Element::Occupied(_, storage_epoch) | Element::Error(storage_epoch, ..) => {
+                    epoch == storage_epoch
+                }
+            },
+            None => false,
         }
     }
 
     /// Get a reference to an item behind a potentially invalid ID.
     /// Panics if there is an epoch mismatch, or the entry is empty.
     pub(crate) fn get(&self, id: I) -> Result<&T, InvalidId> {
         let (index, epoch, _) = id.unzip();
-        let (result, storage_epoch) = match self.map[index as usize] {
-            Element::Occupied(ref v, epoch) => (Ok(v), epoch),
-            Element::Vacant => panic!("{}[{}] does not exist", self.kind, index),
-            Element::Error(epoch, ..) => (Err(InvalidId), epoch),
-        };
-        assert_eq!(
-            epoch, storage_epoch,
-            "{}[{}] is no longer alive",
-            self.kind, index
-        );
-        result
+        match self.map.get(index as usize) {
+            Some(element) => {
+                let (result, storage_epoch) = match *element {
```

Comment:
Here you can explicitly return Err in the None branch to avoid the double match here